packages/valory/contracts/conditional_tokens/contract.py

```diff
@@ -30,11 +30,39 @@
 from hexbytes import HexBytes
 
 
+FIVE_MINUTES = 300.0
+
 class ConditionalTokensContract(Contract):
     """The ConditionalTokens smart contract."""
 
     contract_id = PublicId.from_str("valory/conditional_tokens:0.1.0")
 
+    @staticmethod
+    def execute_with_timeout(func: Callable, timeout: float) -> Any:
+        """Execute a function with a timeout."""
+
+        # Create a ProcessPoolExecutor with a maximum of 1 worker (process)
+        with concurrent.futures.ThreadPoolExecutor(max_workers=1) as executor:
+            # Submit the function to the executor
+            future = executor.submit(
+                func,
+            )
+
+            try:
+                # Wait for the result with a 5-minute timeout
+                data = future.result(timeout=timeout)
+            except TimeoutError:
+                # Handle the case where the execution times out
+                err = f"The RPC didn't respond in {timeout}."
+                return None, err
+
+            # Check if an error occurred
+            if isinstance(data, str):
+                # Handle the case where the execution failed
+                return None, data
+
+            return data, None
```

Comment:
This is duplicated on purpose to avoid introducing deps between the contracts. I think it should ultimately be something offered by the ledger connection, as it already is executing these ledger interactions in a ThreadPoolExecutor. 
I think we should extend the current contract_api and ledger_api protocol to include an optional timeout field.